org.metaborg.spoofax.shell.console/src/main/java/org/metaborg/spoofax/shell/client/console/impl/TerminalUserInterface.java

```diff
@@ -81,27 +81,49 @@ protected void saveLine(String lastLine) {
         lines.add(lastLine);
     }
 
-    /**
-     * Set the prompt to display.
-     *
-     * @param promptString
-     *            The prompt string.
-     */
+    // -------------- IEditor --------------
+    @Override
+    public String getInput() {
+        String input = null;
+        String lastLine;
+        reader.setPrompt(ansi(prompt));
+        try {
+            // While the input is not empty, keep asking.
+            while ((lastLine = reader.readLine()) != null && lastLine.trim().length() > 0) {
+                reader.flush();
+                reader.setPrompt(ansi(continuationPrompt));
+                saveLine(lastLine);
+                // Concatenate the strings with newlines in between.
+                input = lastLine == null ? null : lines.stream().collect(Collectors.joining("\n"));
+                // Clear the lines for next input.
+                lines.clear();
+            }
+        } catch (IOException e) {
```

Comment:
I removed the `throws` from the interface because not all implementations deal with `IOException`s when reading; it is therefore a burden to have this for all implementations.